codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -0,0 +1,174 @@
+/**
+ * File: array_hash_map.c
+ * Created Time: 2023-03-18
+ * Author: Guanngxu (446678850@qq.com)
+ */
+
+#include "../include/include.h"
+
+/* 哈希表初始化函数 */
+array_hash_map_t *new_array_hash_map() {
+    array_hash_map_t *array_hash_map = malloc(sizeof(array_hash_map_t));
+    return array_hash_map;
+}
+
+/* 哈希函数 */
+int hash_func(int key) {
+    int index = key % HASH_MAP_DEFAULT_SIZE;
+    return index;
+}
+
+/* 查询操作 */
+const char *get(const array_hash_map_t *d, const int key) {
+    int index = hash_func(key);
+    const entry_t *pair = d->buckets + index;
+    if(pair == NULL) 
+        return NULL;
+    return pair->val;
+}
+
+/* 添加操作 */
+void put(array_hash_map_t *d, const int key, const char *val) {
+    int index = hash_func(key);
+    d->buckets[index].key = key;
+    d->buckets[index].val = malloc(strlen(val) + 1);
+    strcpy(d->buckets[index].val, val);
+}
+
+/* 删除操作 */
+void remove_item(array_hash_map_t *d, const int key) {
+    int index = hash_func(key);
+    free(d->buckets[index].val);
+    d->buckets[index].val = NULL;
+}
```

Comment:
When the underlying storage structure is an array, in my opinion, the meaning of the delete operation here is only to release the value under the corresponding key, not to delete the `entry_t` structure.